src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -55,11 +55,10 @@ export default function SavedObjectFactory(es, kbnIndex, Promise, Private, Notif
     let customInit = config.init || _.noop;
 
     // optional search source which this object configures
-    self.searchSource = config.searchSource && new SearchSource();
+    self.searchSource = config.searchSource ? new SearchSource() : null;
```

Comment:
for me both `null` and `undefined` are "nothing". 

Sure we can make this more "sophisticated" and say that "nothing" is "something" and therefore there are different types of "nothing" - a "nothing" that is really a something (a value) and a nothing that is really a nothing and therefore we're not sure what that nothing is because... well.. it's nothing. Then we can have a discussion around whether an undefined argument should really be a nothing or a something, and whether all the optional args should have a default value of "real" nothing - `null`... otherwise they'll be really nothing, therefore unknown... and what do you do with unknown? is there a special treatment for unknown vs. known but nothing? (/me taking a break to consult Descartes's essays...)

alternatively, we can just say that nothing is nothing, and both `null` and `undefined` are nothing and we can just choose one of them for the sake of simplicity. And since undefined is more ingrained in the language (e.g. optional arguments and uninitialised vars will be undefined by default) we'll just use `undefined` and ban the use of `null`... just another existential thought...